R/CompDbSource.R

```diff
@@ -221,196 +275,264 @@ setClass("MetFragParam",
-             if (length(object@fragmentpeakmatchabsolutemassdeviation) > 1 || object@fragmentpeakmatchabsolutemassdeviation < 0)
-                 msg <- c("'fragmentpeakmatchabsolutemassdeviation' has to be a positive number of length 1")
-             if (length(object@fragmentpeakmatchrelativemassdeviation) > 1 || object@fragmentpeakmatchrelativemassdeviation < 0)
-                 msg <- c("'fragmentpeakmatchrelativemassdeviation' has to be positive number of length 1")
-              if (length(object@databasesearchrelativemassdeviation) > 1 || object@databasesearchrelativemassdeviation < 0)
-                 msg <- c("'databasesearchrelativemassdeviation' has to be positive number of length 1")
+			 if (!all(unlist(object@validAdducts) %in%
+                      c(adductNames("negative"), adductNames("positive"))
+                      ))
+                 msg <- paste0("'validAdducts' contains unrecognized adduct ",
+                               "names. Check 'MetaboCoreUtils::adductNames' ",
+                               "documentation")
+             if (length(object@fragmentpeakmatchabsolutemassdeviation) > 1 ||
+                 object@fragmentpeakmatchabsolutemassdeviation < 0)
+                 msg <- c(msg,
+                          paste0("'fragmentpeakmatchabsolutemassdeviation'",
+                                 " has to be a positive number of length 1"))
+             if (length(object@fragmentpeakmatchrelativemassdeviation) > 1 ||
+                 object@fragmentpeakmatchrelativemassdeviation < 0)
+                 msg <- c(msg,
+                          paste0("'fragmentpeakmatchrelativemassdeviation' has",
+                                 " to be positive number of length 1"))
+              if (length(object@databasesearchrelativemassdeviation) > 1 ||
+                  object@databasesearchrelativemassdeviation < 0)
+                 msg <- c(msg,
+                          paste0("'databasesearchrelativemassdeviation' has ",
+                                 "to be positive number of length 1"))
              msg <- c(msg, .valid_threshfun(object@THRESHFUN))
              if (any(object@toleranceRt < 0))
-                 msg <- c("'toleranceRt' has to be positive")
+                 msg <- c(msg, "'toleranceRt' has to be positive")
              if (any(object@percentRt < 0))
-                 msg <- c("'percentRt' has to be positive")
+                 msg <- c(msg, "'percentRt' has to be positive")
 			 if (object@minFragCount < 3)
-				msg <- c("Minimum number of input spectra fragments needs to be at least three")
+				msg <- c(msg, paste0("Minimum number of input spectra ",
+                                     "fragments needs to be at least three"))
              msg
          })
 
-#' Default constructor function for 'MetFragParam'
+#' @rdname MetFrag
+#'
+#' @export
 MetFragParam <- function(...) {
```

Comment:
here we need later to allow the user to specify and provide settings for MetFrag.